MergeConflictException while running @nrwl/angular:move schematic

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->
A MergeConflictException is thrown while trying to rename a library.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->
The library should be renamed without error.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

Generate a library:
```
 npm run nx -- generate @nrwl/angular:library mylib
```
Try to rename the library:
```
npm run nx -- generate @nrwl/angular:move --project mylib otherlib
```


<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->


### Failure Logs
<!-- Please include any relevant log snippets or files here. -->
```
MergeConflictException [Error]: A merge conflicted on path "/libs/mylib".
    at /Users/snieuwen/Git/foobar/node_modules/@angular-devkit/schematics/src/tree/host-tree.js:182:31
    at Array.forEach (<anonymous>)
    at HostTree.merge (/Users/snieuwen/Git/foobar/node_modules/@angular-devkit/schematics/src/tree/host-tree.js:117:23)
    at MapSubscriber.project (/Users/snieuwen/Git/foobar/node_modules/@nrwl/angular/node_modules/@angular-devkit/schematics/src/rules/schematic.js:27:19)
    at MapSubscriber._next (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/operators/map.js:49:35)
    at MapSubscriber.Subscriber.next (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/Subscriber.js:66:18)
    at ThrowIfEmptySubscriber._next (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/operators/throwIfEmpty.js:44:26)
    at ThrowIfEmptySubscriber.Subscriber.next (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/Subscriber.js:66:18)
    at TakeLastSubscriber._complete (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/operators/takeLast.js:71:29)
    at TakeLastSubscriber.Subscriber.complete (/Users/snieuwen/Git/foobar/node_modules/rxjs/internal/Subscriber.js:78:18)
A merge conflicted on path "/libs/mylib".
npm ERR! code 1
npm ERR! path /Users/snieuwen/Git/foobar
npm ERR! command failed
npm ERR! command sh -c nx "generate" "@nrwl/angular:move" "--project" "mylib" "otherlib"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/snieuwen/.npm/_logs/2021-03-17T21_43_36_663Z-debug.log
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
```
  Node : 14.16.0
  OS   : darwin x64
  npm  : 7.6.2
  
  nx : Not Found
  @nrwl/angular : 11.5.1
  @nrwl/cli : 11.5.1
  @nrwl/cypress : 11.5.1
  @nrwl/devkit : 11.5.1
  @nrwl/eslint-plugin-nx : 11.5.1
  @nrwl/express : Not Found
  @nrwl/jest : 11.5.1
  @nrwl/linter : 11.5.1
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.5.1
  @nrwl/web : Not Found
  @nrwl/workspace : 11.5.1
  @nrwl/storybook : Not Found
  @nrwl/gatsby : Not Found
  typescript : 4.0.7
```

--- a/packages/angular/src/schematics/move/move.spec.ts
+++ b/packages/angular/src/schematics/move/move.spec.ts
@@ -1,0 +1,38 @@
+import { Tree } from '@angular-devkit/schematics';
+import { createEmptyWorkspace } from '@nrwl/workspace/testing';
+
+import { runSchematic } from '../../utils/testing';
+
+import { Schema } from './schema';
+
+describe('@nrwl/angular:move', () => {
+  let tree: Tree;
+
+  beforeEach(async () => {
+    tree = createEmptyWorkspace(Tree.empty());
+
+    tree = await runSchematic(
+      'library',
+      {
+        name: 'mylib',
+      },
+      tree
+    );
+  });
+
+  it('should move a project', async () => {
+    const result = await runSchematic<Schema>(
+      'move',
+      {
+        projectName: 'mylib',
+        destination: 'mynewlib',
+        updateImportPath: true,
+      },
+      tree
+    );
+
+    expect(result.exists('libs/mynewlib/src/lib/mynewlib.module.ts')).toEqual(
+      true
+    );
+  });
+});

--- a/packages/tao/src/shared/tree.ts
+++ b/packages/tao/src/shared/tree.ts
@@ -143,10 +143,7 @@ export class FsTree implements Tree {
     };
 
     // Delete directories when
-    if (
-      this.exists(dirname(this.rp(filePath))) &&
-      this.children(dirname(this.rp(filePath))).length < 1
-    ) {
+    if (this.children(dirname(this.rp(filePath))).length < 1) {
       this.delete(dirname(this.rp(filePath)));
     }
   }

--- a/packages/workspace/src/generators/move/lib/move-project.ts
+++ b/packages/workspace/src/generators/move/lib/move-project.ts
@@ -22,6 +22,4 @@ export function moveProject(
     tree.write(newFilePath, content);
     tree.delete(file);
   });
-
-  tree.delete(project.root);
 }

